django/utils/tree.py

```diff
@@ -43,13 +45,13 @@ def _new_instance(cls, children=None, connector=None, negated=False):
 
     def __str__(self):
         if self.negated:
-            return '(NOT (%s: %s))' % (self.connector, ', '.join(str(c) for c
-                    in self.children))
-        return '(%s: %s)' % (self.connector, ', '.join(str(c) for c in
-                self.children))
+            return force_str('(NOT (%s: %s))' % (self.connector, ', '.join(force_text(c) for c
+                    in self.children)))
+        return force_str('(%s: %s)' % (self.connector, ', '.join(force_text(c) for c in
+                self.children)))
```

Comment:
Patch LGTM.

Do you think it might be worth removing the code duplication for the `negated` case while were around? It looks we're simply reusing the same logic wrapped in `'(NOT %s)'`.